src/map/script.cpp

```diff
@@ -8853,6 +8853,136 @@ BUILDIN_FUNC(getequipuniqueid)
+	int32 id = reference_getid(data);
+	int32 idx = reference_getindex(data);
+	struct reg_db *ref = reference_getref(data);
+
+	if (not_server_variable(*name) && !sd) {
+		ShowError("buildin_uniqueid_getiteminfo: Cannot use a player variable '%s' if no player is attached.\n", name);
+		script_reportdata(data);
+		st->state = END;
+		return SCRIPT_CMD_FAILURE;
+	}
+
+	if (is_string_variable(name)) {
+		ShowError("buildin_uniqueid_getiteminfo: Cannot Assign String variable '%s'.\n", name);
+		script_reportdata(data);
+		st->state = END;
+		return SCRIPT_CMD_FAILURE;
+	}
+
+	int i;
+
+	ARR_FIND(0, MAX_INVENTORY, i, sd->inventory.u.items_inventory[i].unique_id == item_uniqueid);
+
+	if (i >= MAX_INVENTORY) {
+		ShowError("buildin_uniqueid_getiteminfo: Item not found (unique_id=%llu).\n", item_uniqueid);
+		script_pushint(st, false);
+		return SCRIPT_CMD_FAILURE;
+	}
+
+	struct item *it = &sd->inventory.u.items_inventory[i];
+	int s = 0;
+
+	set_reg_num(st, sd, reference_uid(id, idx + s++), name, it->nameid, ref);
+	set_reg_num(st, sd, reference_uid(id, idx + s++), name, it->refine, ref);
+	set_reg_num(st, sd, reference_uid(id, idx + s++), name, it->identify, ref);
+	set_reg_num(st, sd, reference_uid(id, idx + s++), name, it->attribute, ref);
+	for (int k = 0; k < MAX_SLOTS; k++)
+		set_reg_num(st, sd, reference_uid(id, idx + s++), name, it->card[k], ref);
+	set_reg_num(st, sd, reference_uid(id, idx + s++), name, it->expire_time, ref);
+	set_reg_num(st, sd, reference_uid(id, idx + s++), name, it->bound, ref);
+
+	for (int k = 0; k < MAX_ITEM_RDM_OPT; k++) {
+		set_reg_num(st, sd, reference_uid(id, idx + s++), name, it->option[k].id, ref);
+		set_reg_num(st, sd, reference_uid(id, idx + s++), name, it->option[k].value, ref);
+		set_reg_num(st, sd, reference_uid(id, idx + s++), name, it->option[k].param, ref);
+	}
+
+	script_pushint(st, true);
+	return SCRIPT_CMD_SUCCESS;
+}
+
+/*==========================================
+ * Delete the item with the specified unique id
+ * Return true on success or false otherwise
+ * uniqueid_delitem(<item_unique_id>{,<char_id>})
+ *------------------------------------------*/
+BUILDIN_FUNC(uniqueid_delitem)
+{
+	TBL_PC* sd;
+
+	if (!script_charid2sd(3, sd)) {
+		script_pushint(st, false);
+		return SCRIPT_CMD_FAILURE;
+	}
+
+	int64 item_uniqueid = script_getnum64(st, 2);
+
+	if (!item_uniqueid) {
+		ShowError("buildin_uniqueid_delitem: unknown item (unique_id=%llu).\n", script_getnum64(st, 2));
+		script_pushint(st, false);
+		return SCRIPT_CMD_FAILURE;
+	}
+
+	int i;
+
+	ARR_FIND(0, MAX_INVENTORY, i, sd->inventory.u.items_inventory[i].unique_id == item_uniqueid);
+
+	if (i >= MAX_INVENTORY) {
+		ShowError("buildin_uniqueid_delitem: Item not found (unique_id=%llu).\n", item_uniqueid);
+		script_pushint(st, false);
+		return SCRIPT_CMD_FAILURE;
+	}
+
+	struct item *it = &sd->inventory.u.items_inventory[i];
+
+	if (it->equip)
+		pc_unequipitem(sd, i, 3);
+
+	if (buildin_delitem_search(sd, it, 0, 0)) {
```

Comment:
`buildin_delitem_search` doesn't check the unique id and it may delete the wrong item. Maybe pc_delitem would do the job